### DISCUSSION TITLE
[Feature]: Video Scrubbing with Preview Thumbnails

### DISCUSSION THREAD

**Discussion by coofzilla** [TheWidlarzGroup/react-native-video#4981] [Ideas]
### What problem are you trying to solve?

Navigating through a video by scrubbing currently provides no visual cues, which makes it hard to find a specific moment. Improving the user experience by providing visual feedback during scrubbing would make navigation easier.

### Proposed solution

Add video scrubbing with thumbnail previews, so a preview thumbnail is shown while the user scrubs through the video to enhance navigation.

![image](https://github.com/user-attachments/assets/5e904177-e8bd-4913-87f5-3da7f2c0c803)

### Platforms

iOS

### Additional context

**Possible implementation**

For `ios` to support `hls` I believe it would involve implementing https://developer.apple.com/documentation/http-live-streaming/hls-authoring-specification-for-apple-devices#Trick-Play

**Comment by moskalakamil** [comment]
Hi @coofzilla, thanks for sharing an idea! Right now it's possible to show preview on video while seeking thanks to Plugins System. You can read more about how to integrate this feature on this blog https://medium.com/@freeboub/react-native-video-plugin-thumbnail-view-5a4a227299b8. And you see the demo here https://github.com/TheWidlarzGroup/react-native-video/pull/3943

**Comment by coofzilla** [comment]
Is there a roadmap for when `plugins` are going to be "stable"? Also, what is the intention on the possibilities, from what I read this is mainly for `analytics` features? but, perhaps I'm misinterpreting this?  

I was imagining something where the community can create plugins for `niche` features and then this way add them as we need it. Is that the direction these plugins are going?

**Comment by freeboub** [comment]
@coofzilla the api is marked as unstable because we don't have much feedback on the implementation. So I hope it will not change. Of course, you can use it and also use it for a thumbnail preview. I wrote a POC with an explanation here : https://medium.com/@freeboub/react-native-video-plugin-thumbnail-view-5a4a227299b8

I hope it helps !

**Comment by freeboub** [comment]
@coofzilla I am also thinking of creating a plug-in to synchronize 2 players (linked to your other feature request). I can elaborate more if you want details on my idea, but unfortunately, I don't have time to implement it ...

**Comment by coofzilla** [comment]
> @coofzilla I am also thinking of creating a plug-in to synchronize 2 players (linked to your other feature request). I can elaborate more if you want details on my idea, but unfortunately, I don't have time to implement it ...

yeah I'd love to hear about it! I haven't had a chance to dive into plugins for `react-native-video` ; so, I'm curious how they work and whats possible with them.

**Comment by matgren** [comment]
@coofzilla have you seen https://x.com/krzysztof_moch/status/1867161660127084759 ? perhaps this is what you were looking for.

**Comment by TWG-AI** [comment]
Thanks for the idea — noted for the team to consider.

> I'm an AI assistant, so double-check any specifics I may have mentioned.
